client/src/app/core/core-services/http-stream-endpoint.service.ts

```diff
@@ -17,8 +15,19 @@ export class HttpStreamEndpointService {
 
     public constructor(private http: HttpService) {}
 
-    public registerEndpoint(name: string, url: string, healthUrl: string, method: HTTPMethod = HTTPMethod.GET): void {
-        this.endpointConfigurations[name] = { url, healthUrl, method };
+    public registerEndpoint(name: string, configuration: EndpointConfiguration): void;
+    public registerEndpoint(name: string, url: string, healthUrl: string, method: HTTPMethod): void;
+    public registerEndpoint(
+        name: string,
+        configuration: EndpointConfiguration | string,
+        healthUrl?: string,
+        method: HTTPMethod = HTTPMethod.GET
+    ): void {
+        if (configuration instanceof EndpointConfiguration) {
+            this.endpointConfigurations[name] = configuration;
+        } else {
+            this.endpointConfigurations[name] = { url: configuration, healthUrl, method };
+        }
```

Comment:
why not simply have
`public registerEndpoint(name: string, configuration: EndpointConfiguration): void` ?
Overloading every possibility seems unnecessary here.